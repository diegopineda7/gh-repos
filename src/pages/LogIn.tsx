import { Formik, Form } from 'formik';
import * as Yup from 'yup';

// Components
import { CustomTextInput } from '../components';

// Hooks
import useAuth from '../hooks/useAuth';
import { REQUIRED_FIELD } from '../constants';

export const LogInPage = () => {
  const { userError, validateUser } = useAuth();

  const validationSchema = Yup.object({
    username: Yup.string().required(REQUIRED_FIELD),
  });

  return (
    <div>
      <h1>Log in</h1>

      <Formik
        initialValues={{
          username: '',
        }}
        onSubmit={validateUser}
        validationSchema={validationSchema}
      >
        {({ isValid, dirty }) => (
          <Form>
            <CustomTextInput
              label="GitHub username"
              name="username"
              placeholder="myuser"
            />

            <button type="submit" disabled={!isValid || !dirty}>
              Log in
            </button>
            <h4>{userError}</h4>
          </Form>
        )}
      </Formik>
    </div>
  );
};
