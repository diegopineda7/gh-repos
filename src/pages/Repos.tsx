import '../styles/styles.css';

export const ReposPage = () => {
  return (
    <div>
      <h1>My repos</h1>
    </div>
  );
};
