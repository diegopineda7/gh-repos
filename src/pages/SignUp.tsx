// Hooks
import useLocalStorage from '../hooks/useLocalStorage';

export const SignUpPage = () => {
  const { userError, askForAccess } = useLocalStorage();

  return (
    <div>
      <h1>Sign Up</h1>

      <button onClick={askForAccess}>Create account</button>
      <h1>{userError}</h1>
    </div>
  );
};
