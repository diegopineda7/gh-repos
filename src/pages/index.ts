export { SignUpPage } from './SignUp';
export { ReposPage } from './Repos';
export { LogInPage } from './LogIn';
