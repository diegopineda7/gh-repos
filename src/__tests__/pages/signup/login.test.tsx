import React from 'react';
import { LogInPage } from '../../../pages';
import renderer from 'react-test-renderer';

const mockedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => ({
    navigate: mockedNavigate,
  }),
}));

describe('Test LogIn page', () => {
  it('render component <LogIn />', () => {
    const tree = renderer.create(<LogInPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
