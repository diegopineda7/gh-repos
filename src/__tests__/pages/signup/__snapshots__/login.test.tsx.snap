// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LogIn page render component <LogIn /> 1`] = `
<div>
  <h1>
    Log in
  </h1>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="text-input-container"
    >
      <label
        htmlFor="username"
      >
        GitHub username
      </label>
      <input
        className="text-input "
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="myuser"
        type="text"
        value=""
      />
    </div>
    <button
      disabled={true}
      type="submit"
    >
      Log in
    </button>
    <h4 />
  </form>
</div>
`;
